vpr/src/place/place_util.cpp

```diff
@@ -462,3 +463,71 @@ void set_block_location(ClusterBlockId blk_id, const t_pl_loc& location) {
     place_ctx.grid_blocks[location.x][location.y].blocks[location.sub_tile] = blk_id;
     place_ctx.grid_blocks[location.x][location.y].usage++;
 }
+
+bool macro_can_be_placed(t_pl_macro pl_macro, t_pl_loc head_pos, bool enable_analytic_placer) {
+    auto& device_ctx = g_vpr_ctx.device();
+    auto& place_ctx = g_vpr_ctx.placement();
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+
+    //Get block type of head member
+    ClusterBlockId blk_id = pl_macro.members[0].blk_index;
+    auto block_type = cluster_ctx.clb_nlist.block_type(blk_id);
+
+    // Every macro can be placed until proven otherwise
+    bool mac_can_be_placed = true;
+
+    // Check whether all the members can be placed
+    for (size_t imember = 0; imember < pl_macro.members.size(); imember++) {
+        t_pl_loc member_pos = head_pos + pl_macro.members[imember].offset;
+
+        //Check that the member location is on the grid
+        if (!is_loc_on_chip(member_pos.x, member_pos.y)) {
+            mac_can_be_placed = false;
+            break;
+        }
+
+        /*
+         * analytical placement approach do not need to make sure whether location could accommodate more blocks
+         * since overused locations will be spreaded by legalizer afterward.
+         * floorplan constraint is not supported by analytical placement yet, 
+         * hence, if macro_can_be_placed is called from analytical placer, no further actions are required. 
+         */
+        if (enable_analytic_placer) {
```

Comment:
I'd delete this continue and the passing of enable_analytic_placer in here.  It's OK for the analytic placer not to support floorplan constraints for now, but that's best checked with a command line compatibility check, not by putting special skip code in lower-level routines.